File: packages/android_intent_plus/lib/put_classes/put_int.dart
import 'package:android_intent_plus/put_classes/base/put_base.dart';
import 'package:json_annotation/json_annotation.dart';

part 'put_int.g.dart';

@JsonSerializable()
class PutInt extends PutBase {
  factory PutInt.create({
    required String key,
    required int value,
  }) =>
      PutInt(
        key: key,
        javaClass: 'PutInt',
        value: value,
      );

  PutInt({
    required String key,
    required String javaClass,
    required this.value,
  }) : super(
          key: key,
          javaClass: javaClass,
        );
  final int value;

  factory PutInt.fromJson(Map<String, dynamic> json) => _$PutIntFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PutIntToJson(this);
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutInt.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;
import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;
import org.json.JSONException;
import org.json.JSONObject;

public class PutInt extends PutBase<Integer> {

  public static final String JAVA_CLASS = "PutInt";

  public PutInt(String key, Integer value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutInt putInt) {
    bundle.putInt(putInt.key, putInt.value);
  }

  public static PutInt fromJson(JSONObject jsonObject) throws JSONException {
    return new PutInt(jsonObject.getString(Constants.KEY), jsonObject.getInt(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}
